src/_guides/language/effective-dart/design.md

```diff
@@ -1045,48 +1040,43 @@ confusion, we avoid saying "untyped" and instead use the following terminology:
 In other words, whether some code is annotated or inferred is orthogonal to
 whether it is `dynamic` or some other type.
 
+Inference is a powerful tool to spare you the effort of writing and reading
+types that are obvious or uninteresting. Omitting types in obvious cases also
+draws the reader's attention to explicit types when those types are important,
+for things like casts.
 
-### DO annotate when Dart infers the wrong type.
+Explicit types are also a key part of robust, maintainable code. They define the
+static shape of an API. They document and enforce what kinds of values are
+allowed to reach different parts of the program.
 
-Sometimes, Dart infers a type, but not the type you want. For example, you may
-want a variable's type to be a supertype of the initializer's type so that you
-can later assign some other sibling type to the variable:
+The guidelines here strike the best balance we've found between brevity and
+explicitness, flexibility and safety. When deciding which types to write, you
+need to answer two questions:
 
-{:.good-style}
-<?code-excerpt "misc/lib/effective_dart/design_good.dart (inferred-wrong)"?>
-{% prettify dart %}
-num highScore(List<num> scores) {
-  num highest = 0;
-  for (var score in scores) {
-    if (score > highest) highest = score;
-  }
-  return highest;
-}
-{% endprettify %}
+* What types should I write because I think it's best for them to be visible in
+  the code?
 
-{:.bad-style}
-<?code-excerpt "misc/lib/effective_dart/design_bad.dart (inferred-wrong)" replace="/ +\/\/ ignore: invalid_assignment\n//g"?>
-{% prettify dart %}
-num highScore(List<num> scores) {
-  var highest = 0;
-  for (var score in scores) {
-    if (score > highest) highest = score;
-  }
-  return highest;
-}
-{% endprettify %}
+* Which types should I write because inference can't provide them for me?
 
-Here, if `scores` contains doubles, like `[1.2]`, then the assignment to
-`highest` will fail since its inferred type is `int`, not `num`. In these cases,
-explicit annotations make sense. In the guidelines below, treat *inferred* as
-meaning *inferred to be the type you want*.
+These guidelines help you answer the first question:
+
+* [PREFER type annotating public top-level variables and fields if the type isn't obvious.](#prefer-type-annotating-public-top-level-variables-and-fields-if-the-type-isnt-obvious)
+* [CONSIDER type annotating private top-level variables and fields if the type isn't obvious.](#consider-type-annotating-private-top-level-variables-and-fields-if-the-type-isnt-obvious)
```

Comment:
For both of these guidelines I think that we should write "fields" before "top-level variables". E.g.,

> PREFER type annotating public fields and top-level variables if ...